edc-tests/e2e-tests/src/test/java/org/eclipse/tractusx/edc/tests/TransferOpenTest.java

```diff
@@ -0,0 +1,164 @@
+import org.eclipse.edc.spi.types.domain.HttpDataAddress;
+import org.eclipse.tractusx.edc.lifecycle.MultiRuntimeTest;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.tractusx.edc.policy.PolicyHelperFunctions.businessPartnerNumberPolicy;
+
+@EndToEndTest
+public class TransferOpenTest extends MultiRuntimeTest {
+
+    private final long ONE_WEEK = 60 * 60 * 24 * 7;
+    MockWebServer server = new MockWebServer();
+
+    ObjectMapper mapper = new ObjectMapper();
+
+
+    @Test
+    @DisplayName("Verify that the callbacks are invoked when opening a transfer")
+    void openTransfer_shouldInvokeCallbacks() throws IOException {
+
+        var expectedEvents = List.of(
+                createEvent(ContractNegotiationInitiated.class),
+                createEvent(ContractNegotiationRequested.class),
+                createEvent(ContractNegotiationAgreed.class),
+                createEvent(ContractNegotiationFinalized.class),
+                createEvent(TransferProcessInitiated.class),
+                createEvent(TransferProcessProvisioned.class),
+                createEvent(TransferProcessRequested.class),
+                createEvent(TransferProcessStarted.class),
+                createEvent(TransferProcessCompleted.class));
+
+        var assetId = "api-asset-1";
+        var url = server.url("/mock/api");
+        server.start();
+
+        var authCodeHeaderName = "test-authkey";
+        var authCode = "test-authcode";
+        plato.createAsset(assetId, Map.of(), HttpDataAddress.Builder.newInstance()
+                .contentType("application/json")
+                .baseUrl(url.toString())
+                .authKey(authCodeHeaderName)
+                .authCode(authCode)
+                .build());
+        plato.createPolicy(businessPartnerNumberPolicy("policy-1", sokrates.getBpn()));
+        plato.createPolicy(businessPartnerNumberPolicy("policy-2", sokrates.getBpn()));
+        plato.createContractDefinition(assetId, "def-1", "policy-1", "policy-2", ONE_WEEK);
+
+        var callbacks = List.of(CallbackAddressDto.Builder.newInstance()
+                .uri(url.toString())
+                .events(Set.of("contract.negotiation", "transfer.process"))
+                .transactional(true)
+                .build());
+
+
+        expectedEvents.forEach(_event -> {
+            server.enqueue(new MockResponse());
+        });
+
+
+        sokrates.openTransfer(plato, assetId, callbacks);
+
+        var events = expectedEvents.stream()
+                .map(this::waitForEvent)
+                .collect(Collectors.toList());
+
+        assertThat(expectedEvents).usingRecursiveFieldByFieldElementComparator().containsAll(events);
```

Comment:
Yeah for now we cannot test the EDR token since we need some stuff upstream. But the EDR will come with events so once on the upstream it's implemented we will have another PR and test the EDR